### DISCUSSION TITLE
Frameless Modal

### DISCUSSION THREAD

**Discussion by flashmediasolutions** [filamentphp/filament#7118] [Help]
Is there a way to create a modal without the white frame encapsulating the header, content, footer of a modal?

```
Tables\Actions\Action::make('Testing')
      ->icon('heroicon-o-refresh')
      ->button()
      ->action(fn ($record) => $record->advance())
      ->modalContent(fn ($record) => view('filament.resources.testing', ['record' => $record])),
```

If I can't, any advice on how I can implement a custom modal for frameless scenarios?

![Screenshot 2023-07-21 at 11 41 18 AM](https://github.com/filamentphp/filament/assets/12239388/00562ad9-717b-415c-b5a4-467f0de8faf1)
![Screenshot 2023-07-21 at 11 43 33 AM](https://github.com/filamentphp/filament/assets/12239388/92116ccd-39e6-48e9-aeff-1fc7de369fb0)

**Comment by pboivin** [comment]
There are a few things you can do to hide some elements like `->modalHeading('')` and `->modalActions([])`, but I think your best bet for a completely frameless modal is to add custom CSS.

**Comment by divdax** [comment]
Using different modal templates would be nice to display different content types.

**Reply by flashmediasolutions** [reply]
Do you know how would I go about overriding the modal template from the default?

**Comment by micraux** [comment]
You could create a new component or:

1. hide your unwanted elements as @pboivin said:
https://filamentphp.com/docs/3.x/actions/modals#customizing-the-modals-heading-description-and-submit-action-label

2. modify your app.css: `.fi-modal-window > div { @apply p-0 !important }`
